### PR TITLE
Calibration bindings.

### PR DESCRIPTION
**Pull request by segalmax**
Calibration related bindings.

### CODE REVIEW

**Comment by yosefm** [comment]
On first glance, you brought over too many files from the C testing_fodder, which you don't use in the Python tests.

**Comment by yosefm** [comment]
The test, in general, should be broken down into smaller tests - one for instantiation, one for set/get pos, etc. Not all the asserts in one test. Think of it like that: sometime you want to run just a single test to test a single feature, you don't want to run the kitchen sink just for that.

**Comment by segalmax** [comment]
There is plenty of inseparable logic in calibration test. Can I (this time) leave it as it is? Anyway, from now on I will try to change my thinking on this issue.

File: py_bind/test/test_calibration_binding.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            os.makedirs(output_directory)
+            
+        # write calibration parameters to files
+        calib_obj.write_calibration(test_output_ori_file_name, test_output_add_file_name)
+        
+        # open the input and output files an compare them
+        output_ori_file = open(test_output_ori_file_name, "r")
+        output_add_file = open(test_output_add_file_name, "r")
+        
+        input_ori_file = open(test_output_ori_file_name, "r")
```
Comment (yosefm) [inline py_bind/test/test_calibration_binding.py]: Wrong file name? shouldn't it be input_ori_file_name?

File: py_bind/test/test_calibration_binding.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        self.assertRaises(ValueError, calib_obj.set_pos, numpy.array([1, 2, 3, 4]))
+        self.assertRaises(ValueError, calib_obj.set_pos, numpy.array([1, 2]))
+        
+        # set angles and assert the angles were set correctly
+        dmatrix_before = calib_obj.get_rotation_matrix()  # dmatrix before setting angles
+        angles_np = numpy.array([0.1111, 0.2222, 0.3333])
+        calib_obj.set_angles(angles_np)
+        dmatrix_after = calib_obj.get_rotation_matrix()  # dmatrix after setting angles
+        # make sure the angles are as were set  
+        numpy.testing.assert_array_equal(calib_obj.get_angles(), angles_np)
```
Comment (yosefm) [inline py_bind/test/test_calibration_binding.py]: This doesn't test anything. If the matrix didn't change, the function possibly did nothing or worked on a scratch meory that didn't survive. You should change the angles to something where the answer is easy to calculate, like (pi/2, 0, 0), and check that you get the correct numbers. I'll send you an example soon.

File: py_bind/test/test_calibration_binding.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        
+        # create an output directory
+        if not os.path.exists(output_directory):
+            os.makedirs(output_directory)
+            
+        # write calibration parameters to files
+        calib_obj.write_calibration(test_output_ori_file_name, test_output_add_file_name)
+        
+        # open the input and output files an compare them
+        output_ori_file = open(test_output_ori_file_name, "r")
```
Comment (yosefm) [inline py_bind/test/test_calibration_binding.py]: These are never used later.

**Comment by yosefm** [comment]
Looks good. I'll propose to upstream.

**Pull request closed without merging**